### PR TITLE
Feature/update cart item

### PR DESCRIPTION
**Pull request by stevenbister**
- Closes: #950 
- Closes: #948 

## 📝 Description

Adds the functionality to update the product variant from within the cart.

## ⛳️ Current behavior (updates)

N/a

## 🚀 New behavior

- Adds product query to get variant id
- Adds mutation and function to update the line item
- Adds fetcher form around the cart card options which posts on change
- Updates the array that handles the pending line items to include updates

### CODE REVIEW

File: packages/osc-ecommerce/app/routes/cart.tsx [hunk 1]
```diff
@@ -124,6 +129,27 @@ export const action: ActionFunction = async ({ request, context }: ActionArgs) =
...
             break;
 
+        case CartAction.UPDATE_CART:
+            const updateLinesIds = formData.get('linesIds')
+                ? (JSON.parse(String(formData.get('linesIds'))) as CartType['id'][])
+                : ([] as CartType['id'][]);
+            const productId = String(formData.get('productId'));
+            const selectedOptions = JSON.parse(String(formData.get('selectedOptions')));
+
+            invariant(updateLinesIds.length, 'No lines to update');
```
Comment (mattwendzina) [inline packages/osc-ecommerce/app/routes/cart.tsx:139]: Same as with remove PR, I tried simulating this invariant and it doesn't look like the errors get caught anywhere, they just crash the app:
Screenshot 2023-06-02 at 11 44 51

**Review by mattwendzina — COMMENTED** [review]
Looks good mate 👌 

Just one comment, similar to the remove PR about the error thrown by `invariant` not being caught anywhere.

File: packages/osc-ecommerce/app/routes/cart.tsx (hunk 1 above)
Comment (stevenbister) [reply packages/osc-ecommerce/app/routes/cart.tsx]: Sorted 👍

**Review by mattwendzina — APPROVED** [review]
LGTM

**Pull request merged**